app/change-password/page.tsx

```diff
@@ -0,0 +1,94 @@
+"use client";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { useSearchParams } from "next/navigation";
+
+const ResetPassword: React.FC = () => {
```

Comment:
Same comment here: wouldn't it be better to componentize this function rather than making it a new page?